deid/dicom/fields.py

```diff
@@ -59,12 +61,13 @@ def name_contains(self, expression):
         self.element.name: is the human friendly name "Sequence Child"
         self.element.keyword: is the name without nesting "Child"
         """
+        regexp_expression = f"^{expression}$" if whole_string else expression
```

Comment:
If the expression is expected to be a regular expression, would the caller not just be able to provide the start and end directives?